Don't bundle extra dependencies

a14m opened the pull request:
Move the requiring of the `oauth`/`oauth2` to the `external` submodule

Add a warning when the dependencies are not there in while generating the migration and the including the external module to notify the user about installing the gems

The warning only shows if the `oauth` _or_ `oauth2` are not bundled when initializing or requiring the `external` submodule

Know issues: if the user installed only `oauth2` (cause he doesn't need `oauth`) the warning still shows.
one solution to this problem is to move the requiring code inside each provider (but that will add a lot of duplicate code) and giving the user the ability to load specific providers (which is just a lot of work without any added benefit, since there is no problem in requiring a provider and never using it)

Solves: #773

Commit:
FEAT: move oauth dependencies to development

Add a warning when installing the "external" submodule to notify the
user about installing the desired gem for use (whether oauth or oauth2)

Solves: #773

Ch4s3 commented:
@arnvald I think we should tag this for 1.0 as its a breaking change.

mad-raz commented:
yeah I think that too... and I'm waiting for the approval to update the changelog

mad-raz commented:
This introduced a bug when running `rails generate sorcery:install` so it's currently not safe to merge... will fix it and reopen it later

Commit:
FIX: rails generator fails because oauth is not bundled

Move the requiring of the oauth to the external submodule

Add a warning when the dependencies are not there in while generating
the migration and the including the external module to notify the user
about installing the gems

The warning only shows if the oauth and oauth2 are not bundled when
initializing or requiring the external submodule

Know issues: if the user installed only oauth2 (cause he doesn't need
oauth) the warning still shows. one solution to this problem is to move
the requiring code inside each provider (but that will add a lot of
duplicate code) and giving the user the ability to load specific
providers (which is just a lot of work without much added benefit)

arnvald commented:
How about we also introduce warning to 0.9.x branch to add these dependencies to Gemfile, so that people can see it before updating to 1.0?

mad-raz commented:
good idea... and add a table of `provider <---> oauth` version mapping so that the users don't bundle both oauth gems when they are going to use only 1

mad-raz commented:
sounds good... but I prefer if we added this warning to the readme file... or do you think that we should add another PR to hard-code this warning on bundling the gem and merge it to the 0-9-x branch ?

arnvald commented:
I suggest following

* add depreciation warning to code and merge to 0-9-x branch
* add `upgrading to 1.x` section to readme and mention it there

mad-raz commented:
will add the warning to a different PR (and will upgrade the readme as part of this PR later)

Commit:
DOCS: update the README about upgrading to 1.0 and external dependencies
